stl/inc/utility

```diff
@@ -927,29 +927,14 @@ _EXPORT_STD [[noreturn]] __forceinline void unreachable() noexcept /* strengthen
 #endif // defined(_DEBUG)
 }
 
+template <class _Ty, class _Uty,
+    class _Tmp = _Maybe_const<is_const_v<remove_reference_t<_Ty>>, remove_reference_t<_Uty>>>
+using _Forward_like_t = conditional_t<is_rvalue_reference_v<_Ty&&>, _Tmp&&, _Tmp&>;

```

Comment:
Given `forward_like` becomes SFINAE-friendly on `_Ty` now, perhaps we can move `_Can_reference` back to `__msvc_iter_core.hpp` and drop the workaround for EDG (which reverts some changes in #2974).

Reply:
Good catch! cleanup applied.